[Failed test] [cmd/opampsupervisor] TestSupervisorBootstrapsCollector/Without_service.AllowNoPipelines failed on Windows

### Component(s)

cmd/opampsupervisor

### Describe the issue you're reporting

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/14356460322/job/40248603867?pr=39271

```
=== RUN   TestSupervisorBootstrapsCollector/Without_service.AllowNoPipelines
    e2e_test.go:671: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/opampsupervisor/e2e_test.go:671
        	Error:      	Received unexpected error:
        	            	fork/exec ./testdata/collector/no_pipelines_fg.sh: %1 is not a valid Win32 application.
        	Test:       	TestSupervisorBootstrapsCollector/Without_service.AllowNoPipelines
--- FAIL: TestSupervisorBootstrapsCollector (0.71s)
    --- PASS: TestSupervisorBootstrapsCollector/With_service.AllowNoPipelines (0.71s)
    --- FAIL: TestSupervisorBootstrapsCollector/Without_service.AllowNoPipelines (0.00s)
```

--- a/cmd/opampsupervisor/e2e_test.go
+++ b/cmd/opampsupervisor/e2e_test.go
@@ -626,6 +626,8 @@ func TestSupervisorBootstrapsCollector(t *testing.T) {
 		{
 			name: "With service.AllowNoPipelines",
 			cfg:  "nocap",
+			precheck: func(t *testing.T) {
+			},
 		},
 		{
 			name: "Without service.AllowNoPipelines",
@@ -643,6 +645,7 @@ func TestSupervisorBootstrapsCollector(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			tt.precheck(t)
 			agentDescription := atomic.Value{}
 
 			// Load the Supervisor config so we can get the location of

